lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -197,6 +197,8 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
             return new JourneyErrorResponse(
                             JOURNEY_ERROR_PATH, e.getResponseCode(), e.getErrorResponse())
                     .toObjectMap();
+        } finally {
+            auditService.awaitAuditEvents();
```

Comment:
The scary thing about this approach is that if we _don't_ call this, then the lambda runtime will suspend and (potentially) interrupt an in-progress request.

I can't think of a great solution to this beyond 'make sure it's included in all the handlers' and 'never construct a new audit service outside of a handler'.

Reply:
Do you think we'd catch this in the pipeline before any mistakes hit prod?

Reply:
Probably not, which is the scary part :/